### PR TITLE
Update Storage.py to 0.3.0

### PR DESCRIPTION
**Pull request by axsaucedo**
This PR only updates the code in our library, but doesn't update the kfserivng image for our storage.py (that is a separate PR).

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **axsaucedo**

If they are not already assigned, you can assign the PR to them by writing `/assign @axsaucedo` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Mon May 18 17:07:37 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1846/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1846 --build=1

**Comment by seldondev** [comment]
Mon May 18 17:08:57 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1846/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1846 --build=2

**Comment by seldondev** [comment]
@axsaucedo: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint | 34261fe7f9df1eb9e930d15a4c867d0a32bca6b7 | [link]() | `/test lint`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@axsaucedo: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint | 34261fe7f9df1eb9e930d15a4c867d0a32bca6b7 | [link]() | `/test lint`
pr-build | 34261fe7f9df1eb9e930d15a4c867d0a32bca6b7 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Mon May 18 18:05:00 UTC 2020
The logs for [pr-build] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1846/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1846 --build=3

**Comment by seldondev** [comment]
Mon May 18 18:05:08 UTC 2020
The logs for [lint] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1846/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1846 --build=4

File: python/seldon_core/storage.py [hunk 1]
```diff
@@ -244,22 +244,17 @@ def _download_local(uri, out_dir=None):
...
             url.scheme == "https"
             if url.scheme
-            # KFServing uses S3_USE_HTTPS, whereas Seldon was already using
-            # USE_SSL.
-            # To keep compatibility with the storage init layer we support
-            # both, giving priority to USE_SSL.
-            # https://github.com/SeldonIO/seldon-core/pull/827
-            # https://github.com/kubeflow/kfserving/pull/362
-            else bool(getenv("USE_SSL", "S3_USE_HTTPS", "false"))
+            else bool(os.getenv("S3_USE_HTTPS", "true"))
```
Comment (adriangonz) [inline python/seldon_core/storage.py:252]: On our docs we document usage of `USE_SSL` (e.g. https://docs.seldon.io/projects/seldon-core/en/latest/servers/overview.html?highlight=USE_SSL#handling-credentials). But KFServing uses `S3_USE_HTTPS` instead. 

Should we add back the removed bit? 

It's a bit of a pain to have that divergence though. Maybe we should consider `USE_SSL` as deprecated and update our docs (and anything on the operator?) to use `S3_USE_HTTPS`. But that's outside the scope of this PR.

Comment (adriangonz) [reply python/seldon_core/storage.py:252]: I've added an issue with the above: https://github.com/SeldonIO/seldon-core/issues/1847

However, in the meantime, we will need to read from both.

Comment (axsaucedo) [reply python/seldon_core/storage.py:252]: Oh I see, no I do think you're right, let's probably add both!

**Comment by seldondev** [comment]
Tue May 19 09:19:47 UTC 2020
The logs for [pr-build] [5] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1846/5.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1846 --build=5

**Comment by seldondev** [comment]
Tue May 19 09:21:48 UTC 2020
The logs for [lint] [6] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1846/6.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1846 --build=6

**Comment by adriangonz** [comment]
/approve

Looks good!

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/SeldonIO/seldon-core/pull/1846#issuecomment-630710437" title="Approved">adriangonz</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [adriangonz]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**